pkg/logs/consumer/minio.go

```diff
@@ -0,0 +1,170 @@
+package consumer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"strconv"
+
+	"github.com/minio/minio-go/v7"
+	"go.uber.org/zap"
+
+	"github.com/kubeshop/testkube/pkg/log"
+	"github.com/kubeshop/testkube/pkg/logs/events"
+	minioconnecter "github.com/kubeshop/testkube/pkg/storage/minio"
+)
+
+const (
+	defaultBufferSize = 100 * 1024 * 1024 // 100MB
+	defaultWriteSize  = 95 * 1024 * 1024  // 95MB
+)
+
+var _ Consumer = &MinioConsumer{}
+
+type ErrMinioConsumerDisconnected struct {
+}
+
+func (e ErrMinioConsumerDisconnected) Error() string {
+	return "minio consumer disconnected"
+}
+
+type BufferInfo struct {
+	Buffer *bytes.Buffer
+	Part   int
+}
+
+// MinioConsumer creates new MinioSubscriber which will send data to local MinIO bucket
+func NewMinioConsumer(endpoint, accessKeyID, secretAccessKey, region, token, bucket string, ssl bool) *MinioConsumer {
+	c := &MinioConsumer{
+		minioConnecter: minioconnecter.NewConnecter(endpoint, accessKeyID, secretAccessKey, region, token, bucket, ssl, log.DefaultLogger),
+		Log:            log.DefaultLogger,
+		bucket:         bucket,
+		region:         region,
+		disconnected:   false,
+		buffInfos:      make(map[string]BufferInfo),
+	}
+	minioClient, err := c.minioConnecter.GetClient()
```

Comment:
I don't get why you need to overcomplicate it with minioConnecter? - looks like you need minio client only here? This one adds additional level of object hierarchy.